Automattic/jetpack
# Premium Content Block: Added a "Remove block and keep content" and fixed to group transformation

In order to avoid cases where users might want to remove the block but keep the content, we've introduced an alias block settings menu item that acts like an alias for `to core/group` transform.  Furthermore, the PR also contains a fix for `to core/group` transforms that will only move the Subscriber view content into the new group block.



Fixes #22424

#### Changes proposed in this Pull Request:

* Fixed transformation to group
* Added a new menu item only for Premium content Block called "Remove block and keep content" that, when pressed, will transform the block into a core/group. Basically, it's just an alias for to group transforms

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


no
#### Testing instructions:






* For Simple, apply D74487-code on your sandbox
* For Atomic, use Jetpack Beta to switch to this branch
* Go to the editor and insert a new Premium content block (an existing one would also work).
* Click block three-dot setting menu (make sure the Premium content block container is selected, it should have the Subscriber and plan information controls).
* Click on the "Remove block and keep content" menu item
* You should be able to see now that the block has been converted to a group block containing only the content from the subscriber view.
* To test the transform, revert your block back to Premium content (or add another one)
* Click on the transforms menu and select "Group"
* Notice that the block has been converted to a group with the content only from the "Subscriber view".

**Video:**

https://user-images.githubusercontent.com/995955/152653803-f38a592e-5364-432c-9a48-b236ab477578.mov
